splinterd/src/main.rs

```diff
@@ -247,20 +242,22 @@ fn start_daemon(matches: ArgMatches) -> Result<(), UserError> {
         }
     };
 
+    let local_node_registry_location = format!("{}{}", config.state_dir(), "nodes.yaml");
```

Comment:
This should probably use Path stuff to do the join. If state_dir() doesn't have a "/", this would result in an unintended path.

Reply:
Updated all of these path constructors to use `Path` in a new commit